common/lib/xmodule/xmodule/modulestore/tests/django_utils.py

```diff
@@ -279,6 +279,18 @@ def setUp(self):
         OverrideFieldData.provider_classes = None
         super(SharedModuleStoreTestCase, self).setUp()
 
+    def reset_test_case(self):
+        """
+        Manually run tearDownClass/setUpClass again.
+
+        This is so that if you have a mostly read-only course that you're just
+        modifying in one test, you can write `self.reset_test_case()` at the
+        end of that test and reset the state of the world for other tests in
+        the class.
+        """
+        self.tearDownClass()
+        self.setUpClass()
```

Comment:
It seems like this would be handy as a method decorator, so that you could do something like:

    @modifies_courseware
    def this_is_my_test_case(self):
        ...